Ap_Scheduler: initialize _active_loop_rate_hz in loop() and remove check in get_loop_rate_hz()

neo-0007 opened the pull request:
- issue: #30234 
- an extra check was made on every call of get_loop_rate_hz() for _active_loop_rate_hz == 0
- remove extra check and initialize the value of _active_loop_rate_hz in AP_Scheduler::loop()

**Testing**
- succesfully build for sitl
![Screenshot from 2025-06-11 00-07-38](https://github.com/user-attachments/assets/d68c1628-32ff-4488-bea7-4eb8941de3c1)
- tested on sitl
![Screenshot from 2025-06-11 00-27-10](https://github.com/user-attachments/assets/ddbdb9cb-4a3b-4c65-bbf7-2e6188e33e62)

Ryanf55 commented:
Code changes look good. I'll rely on autotest for this.

peterbarker commented:
@neo-0007 please avoid adding Merge commits to your branch.  If you want to udpate it, make sure to select "Update with Rebase".

neo-0007 commented:
Hi @peterbarker , I have cleaned up my branch by removing the merge commits and rebased it onto the latest master as requested. I am now working on applying the suggested changes and will push them shortly!

neo-0007 commented on libraries/AP_Scheduler/AP_Scheduler.cpp:
```diff
@@ -363,6 +363,10 @@ void AP_Scheduler::loop()
 
+    if (_active_loop_rate_hz == 0) {
+       _active_loop_rate_hz = _loop_rate_hz;
+    }
```
I tried removing this code, but the autotest fails without it. Even with the initialization in init() the test still fails. With this line added in loop(), all tests pass. So it seems like it's needed to make sure _active_loop_rate_hz is set early enough.

<img width="1920" height="1080" alt="Screenshot_20250713_011233" src="https://github.com/user-attachments/assets/22089e33-e8ca-4620-a0f0-2a11a517c632" />

<img width="1920" height="1080" alt="Screenshot_20250713_011249" src="https://github.com/user-attachments/assets/5deb8e06-44bb-4a5f-8d44-fda215ad90c8" />

tpwrules replied:
Do you know which autotests fail? I was able to run the CI suite on my fork without the set in `loop()` just fine on top of current master.

tpwrules commented:
#31138 should fix the need for the extra set, but there is some more cleanup to do. Are you interested in further work on this?

neo-0007 commented:
@tpwrules
I implemented the suggested changes and also included some changes previously mentioned by @peterbarker.  

I’ve opened PR #31165 — I’d appreciate a review and any further suggestions. Happy to make adjustments as needed. Thanks!
